src/LinuxTracing/ContextSwitchAndThreadStateVisitor.cpp

```diff
@@ -85,6 +86,16 @@ void ContextSwitchAndThreadStateVisitor::ProcessInitialState(uint64_t timestamp_
 }
 
 void ContextSwitchAndThreadStateVisitor::visit(TaskNewtaskPerfEvent* event) {
+  std::optional<pid_t> new_pid = GetPidOfTid(event->GetNewTid());
+  ThreadName thread_name;
+  if (new_pid.has_value()) {
+    thread_name.set_pid(new_pid.value());
```

Comment:
Maybe make it explicit set_pid(new_pid.value_or(-1)) ?

Also when does it happen? (add a comment?)

Reply:
Done, more in line with context switches.

This happens if something goes wrong with the bookkeeping of pid-tid associations, so hopefully never. I'd rather not add a comment because this is a common theme of the class, not a special case here.